src/js/tracing/timetodisplay.tsx

```diff
@@ -262,7 +262,12 @@ function updateFullDisplaySpan(frameTimestampSeconds: number, passedInitialDispl
     return;
   }
 
-  span.end(frameTimestampSeconds);
+  if (initialDisplayEndTimestamp > frameTimestampSeconds) {
+    logger.warn(`[TimeToDisplay] Using initial display end. Full display end frame timestamp is before initial display end.`);
+    span.end(initialDisplayEndTimestamp);
+  } else {
+    span.end(frameTimestampSeconds);
+  }
 
   span.setStatus('ok');
   logger.debug(`[TimeToDisplay] ${spanToJSON(span).description} span updated with end timestamp.`);
```

Comment:
Not related to this PR:
I noticed on line 273 and also 261 that we are calling `spanToJSON(span).description` twice, ideally we could store this value on a constant if description doesn't change.

Also, When users have the logger disabled, we could avoid calling spanToJSON,  allowing a small performance slight performance improvement by avoiding a call to a json serializer when it's not needed.